packages/delegation-process/src/services/delegationProducerService.ts

```diff
@@ -1,8 +1,112 @@
+import { delegationApi } from "pagopa-interop-api-clients";
+import {
+  AppContext,
+  DB,
+  eventRepository,
+  WithLogger,
+} from "pagopa-interop-commons";
+import {
+  Delegation,
+  delegationEventToBinaryDataV2,
+  DelegationId,
+  delegationKind,
+  delegationState,
+  EServiceId,
+  generateId,
+  Tenant,
+  TenantId,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { delegationNotFound, tenantNotFound } from "../model/domain/errors.js";
+import { toCreateEventProducerDelegation } from "../model/domain/toEvent.js";
 import { ReadModelService } from "./readModelService.js";
+import {
+  assertDelegationNotExists,
+  assertDelegatorIsIPA,
+  assertDelegatorIsNotDelegate,
+  assertEserviceExists,
+  assertTenantAllowedToDelegation,
+} from "./validators.js";
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export function delegationProducerServiceBuilder(
-  _readModelService: ReadModelService
+  dbInstance: DB,
+  readModelService: ReadModelService
 ) {
-  return {};
+  const getTenantById = async (tenantId: TenantId): Promise<Tenant> => {
+    const tenant = await readModelService.getTenantById(tenantId);
+    if (!tenant) {
+      throw tenantNotFound(tenantId);
+    }
+    return tenant;
+  };
+
+  const repository = eventRepository(dbInstance, delegationEventToBinaryDataV2);
+  return {
+    async getDelegationById(delegationId: DelegationId): Promise<Delegation> {
+      const delegation = await readModelService.getDelegationById(delegationId);
+
+      if (!delegation) {
+        throw delegationNotFound(delegationId);
+      }
+
+      return delegation;
+    },
```

Comment:
This is duplicated. Please remove it, we already have this in `delegationService`